modules/dbparser/dbparser-grammar.ym

```diff
@@ -79,11 +79,15 @@ parser_db_opts
 /* NOTE: we don't support parser_opt as we don't want the user to specify a template */
 parser_db_opt
         : KW_FILE '(' string ')'                		{ log_db_parser_set_db_file(((LogDBParser *) last_parser), $3); free($3); }
-	| KW_INJECT_MODE '(' parser_db_inject_mode ')'		{ log_db_parser_set_inject_mode(((LogDBParser *) last_parser), $3); free($3); }
-	| parser_opt
+	| stateful_parser_opt
         ;
 
-parser_db_inject_mode
+stateful_parser_opt
+	: KW_INJECT_MODE '(' stateful_parser_inject_mode ')'	{ stateful_parser_set_inject_mode(((StatefulParser *) last_parser), $3); free($3); }
+	| parser_opt
+	;
+
+stateful_parser_inject_mode
 	: string				{ $$ = $1; }
```

Comment:
Wouldn't it be better to support the `pass-through` inject mode as `KW_PASS_THROUGH`? So the grammar would fail if the inject mode is not `internal` or `pass-through`.